src/draw/sdl/lv_draw_sdl.c

```diff
@@ -224,10 +224,37 @@ static bool draw_to_texture(lv_draw_sdl_unit_t * u, cache_data_t * data)
             break;
         case LV_DRAW_TASK_TYPE_IMAGE: {
                 lv_draw_image_dsc_t * image_dsc = task->draw_dsc;
-                const char * path = image_dsc->src;
-                SDL_Surface * surface = IMG_Load(&path[2]);
+                lv_image_src_t type = lv_image_src_get_type(image_dsc->src);
+                SDL_Surface * surface = NULL;
+                if(type == LV_IMAGE_SRC_FILE) {
+                    const char * path = image_dsc->src;
+                    surface = IMG_Load(&path[2]);
+                }
+                else if(type == LV_IMAGE_SRC_VARIABLE) {
+                    lv_image_dsc_t * lvd = image_dsc->src;
+                    surface = SDL_CreateRGBSurfaceFrom(lvd->data,
+                                                       lvd->header.w, lvd->header.h,
+                                                       LV_COLOR_FORMAT_GET_BPP(lvd->header.cf),
+                                                       lvd->header.stride,
+#if SDL_BYTEORDER == SDL_LIL_ENDIAN
+                                                       0x00FF0000,
+                                                       0x0000FF00,
+                                                       0x000000FF,
+                                                       0xFF000000
+#else
+                                                       0x0000FF00,
+                                                       0x00FF0000,
+                                                       0xFF000000,
+                                                       0x000000FF
+#endif
+                                                      );
+                }
+                else {
+                    LV_LOG_WARN("image source type unknown\n");
+                }
+
                 if(surface == NULL) {
-                    fprintf(stderr, "could not load image: %s\n", IMG_GetError());
+                    LV_LOG_ERROR("could not load image: %s\n", IMG_GetError());
```

Comment:
`IMG_GetError()` belongs to `IMG_Load`.

My suggestion was:
```
if(type == LV_IMAGE_SRC_FILE) {
 ...
 surface = IMG_Load(&path[2]);
 if(surface == NULL) {
   LV_LOG_ERROR(....., IMG_GetError());
   return false;
 }
}
else if(type == LV_IMAGE_SRC_VARIABLE) {
 ...
 surface = SDL_CreateRGBSurfaceFrom(....);
 if(surface == NULL) {
  LV_LOG_ERROR(.....);
  return false;
 }
}
else {
 LV_LOG_WARN(....);
 return false;
}
```

So, when the source is file, then we get more details from `IMG_GetError()`, when source is variable, then we just log the error, and when the source is unknown, the log warning/error, and return.

Reply:
I see, should be all better now.